clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/jobs/ComputeJob.java

```diff
@@ -0,0 +1,46 @@
+package com.linkedin.venice.hadoop.jobs;
+
+import com.linkedin.venice.utils.VeniceProperties;
+import java.io.Closeable;
+import java.io.IOException;
+
+
+/**
+ * An interface to abstract executing and monitoring a compute job running on any batch compute engine
+ */
+public interface ComputeJob extends Closeable {
+  enum Status {
+    NOT_STARTED(0), RUNNING(1), SUCCEEDED(2), FAILED(3), FAILED_VERIFICATION(4), KILLED(5);
+
+    public final int value;
+    private static final Status[] values = values();
+
+    Status(int value) {
+      this.value = value;
+    }
+
+    public static Status valueOf(int value) {
+      return values[value];
+    }
+
+    public boolean isEnded() {
+      return this == SUCCEEDED || this == FAILED || this == FAILED_VERIFICATION || this == KILLED;
```

Comment:
I don't think I need the ordinal. Let me see if it can be removed.

I can do `isTerminal` method, and it works in this case. When writing it originally, I was thinking we'd need other cases like `isRunning`, `isEnded`, etc and thought that maybe the conditional approach would have been better. But now that the only method needed is `isEnded`, I'll do the method you suggested.

Reply:
It looks like the ordinals and the `isEnded` are not used. I'll remove both of these